doc/rfc/0000-rfc-repository.md

```diff
@@ -0,0 +1,168 @@
+# RFC 0000 - RFC repository
+
+- Date proposed: 2023-02-28
+- Date approved: under review
+
+## Summary
+
+This RFC proposes storing OpenTitan RFCs as files in a Git repo, and using
+an open source development workflow for the RFC process. RFCs would be
+stored in the RFC repo as text files. RFCs would be proposed, reviewed, and
+accepted/rejected using GitHub pull requests (PRs). The goal is to ensure
+project decisions are better documented, and to make the RFC process more
+visible and transparent. The workflow proposed here will already be familiar to
+open source contributors and should add little overhead.
+
+## Motivation
+
+RFCs in OpenTitan are used for decisions that need consensus from the technical
+committee. The RFC workflow before this proposal is roughly:
+
+1. Write up your RFC somewhere.
+2. Create an issue labelled `RFC:Proposal` linking to your RFC in the main repo.
+3. Receive and address comments from others.
+4. Apply the `RFC:TC review` label when the RFC is ready to be voted on.
+5. The technical committee will vote on pending RFCs at their next meeting.
+5. Approved RFCs have their label changed to `RFC:Approved`.
+6. Rejected RFCs have their label removed.
+
+Typically, authors write their RFCs in one of two places:
+
+* Google Docs.
+* The GitHub issue's body as Markdown.
+
+The entry-point for finding RFCs stored like this is through the `RFC:Approved`
+label in GitHub issues. The OpenTitan repository gets many issues and has many
+labels to sort them. Discovering RFCs requires regularly checking the `RFC`
+labels or keeping up to date with OpenTitan's fast moving issues.
+
+Allowing authors to choose where they store their RFC content makes it
+harder to reference them as documentation. It also means we can't manage them
+consistently: we have no public-by-default rule for Google Docs, leading to
+many RFCs restricted only to select people. Comments are left in both GitHub
+iAssue comments and Google Docs reviews.
+
+In summary, the motivation of this RFC is to have:
+
+1. a single store of RFC documents,
+2. where RFCs can easily be discovered,
+3. where RFC content can be enforced to be public,
+4. where RFC comments are given and addressed in a consistent workflow.
```

Comment:
Re "RFC content can be enforced to be public" and "RFC comments are given and addressed in a consistent workflow":

I assume this (and the text below) means RFCs get proposed and reviewed exclusively through GitHub PRs that add a Markdown doc to `doc/rfc/`, right? If so, I see two problems:
- While the final content of an RFC is public, the draft content and the comments may contain project-internal information.
- GitHub's GUI for adding comments and suggesting changes to a document is not on par with the suggestions functionality of Google Docs.

Reply:
> While the final content of an RFC is public, the draft content and the comments may contain project-internal information.

There were some comments on private RFCs/comments in a previous draft, but I removed them. Private comments need to be communicated over a private channel, and it's a shame there's no concept of a private comment in GitHub.

One way to solve this would be to have a separate private repo where sensitive RFCs are processed, then committed to the main repo publicly when ready. This is what I suggested in the previous draft when I thought we had private RFCs, however there are some problems:

* Maintaining two repos is a pain.
* You might not know whether your RFC is going to receive private comments when you make it.
* We still want any public comments to be visible, but having a separate repo is all-or-nothing.

For RFCs we think might be potentially sensitive [current guidance](https://opentitan.org/book/doc/project_governance/rfc_process.html#rfc-lifetime) we currently have this rule:

> If the RFC potentially contains information that could be certification-sensitive (guidance to be shared), send a note to security@opentitan.org first for feedback.

Do you think this would be enough to cover RFCs that need private comments? Send an email first if you think it's sensitive, and if it's submitted and private comments are required, those go over some private channel like email?

Reply:
> GitHub's GUI for adding comments and suggesting changes to a document is not on par with the suggestions functionality of Google Docs.

I'm not sure whether I like Google Docs' suggestions any more than GitHub's, but agree that GitHub's isn't perfect. See also the suggestion to review RFCs through Gerrit instead of GitHub: https://github.com/lowRISC/opentitan/pull/17414#issuecomment-1453345636.